### PR TITLE
Add local OS keyring support so we don't have to req password

### PR DESCRIPTION
**Pull request by chmouel**
It allows you to leverage your local OS keyring as supported by the library :

https://github.com/zalando/go-keyring

Example :

```bash
osio login -u username -t preview --keyring-user KEYRING_USER --keyring-service KEYRING_SERVICE
```

on OSX you can add your password to your keychain like this :

```bash
security add-generic-password -a KEYRING_USER -s KEYRING_SERVICE -w 'THE_PASSWORD_!!!_!!!'
```

(you may want to add a space before the cmd or clean your shell history after that)

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -32,4 +32,8 @@
 [[constraint]]
   name = "github.com/fabric8-services/fabric8-common"
   revision = "2efd83d29b1e5c657728e83fe03878569c0f122f"
-  
\ No newline at end of file
+
+[[constraint]]
```
Comment (xcoulon) [inline Gopkg.toml]: why do you add the repository as a self-dependency?

File: cmd/login.go [hunk 2]
```diff
@@ -53,10 +59,17 @@ func login(cmd *cobra.Command, args []string) {
 			username = strings.TrimRight(scanner.Text(), "\r\n")
 		}
 	}
-	fmt.Print("password: ")
-	password, err := terminal.ReadPassword(int(os.Stdin.Fd()))
-	if err != nil {
-		log.Fatal(err)
+
+	// try to get password from keyring, it's okay if we can't find it (i.e: ignore error)
+	password, _ := keyring.Get(keyringUser, keyringService)
```
Comment (xcoulon) [inline cmd/login.go]: this did not work for me: when retrieving and printing the error, I have the following message: `secret not found in keyring`. I double-checked the account and service thatI used when running the `security add-generic-password` command, so I'm a bit lost... (BTW, if you add a space before the command, it won't be recorded in the session history)

**Review by xcoulon — CHANGES_REQUESTED** [review]

File: cmd/login.go (hunk 2 above)
Comment (chmouel) [reply cmd/login.go]: yep sorry I need to send an update, i have miss placed the arguments between service and user,

Comment (xcoulon) [reply cmd/login.go]: doh, of course! I did not think about checking the order of the args in the `keyring.Get()` func call 🤦‍♂️ It works much better now ;)

**Comment by xcoulon** [comment]
@chmouel also, I wonder if we could get rid of the `-u` arg if the keyring account is an email address. WDYT?

File: Gopkg.toml (hunk 1 above)
Comment (chmouel) [reply Gopkg.toml]: There is no https://github.com/fabric8-services/fabric8-auth-cli and that was the only way for me to get it to compile, unless I do something like 



```diff
diff --git a/main.go b/main.go
index a394fdf..04e029e 100644
--- a/main.go
+++ b/main.go
@@ -6,8 +6,8 @@ import (

        "github.com/sirupsen/logrus"

-       "github.com/fabric8-services/fabric8-auth-cli/cmd"
        "github.com/fabric8-services/fabric8-common/log"
+       "github.com/xcoulon/fabric8-auth-cli/cmd"
 )

 func main() {
```

**Comment by chmouel** [comment]
@xcoulon make it implicit between keyring account and OSIO user ? 

I have multiple test users to the same password on OSIO  I can reference all of them to the same keyring key so i can avoid a lot of duplications,

**Comment by xcoulon** [comment]
> @xcoulon make it implicit between keyring account and OSIO user ?
> 
> I have multiple test users to the same password on OSIO I can reference all of them to the same keyring key so i can avoid a lot of duplications,

ok, I see what you mean, but let's say that in case the `-u` flag was missing and the `-keyring-account` value was a valid email address, then we could use the latter as the username, couldn't we? but if you want to specify the username, then that would still work

File: cmd/login.go (hunk 2 above)
Comment (chmouel) [reply cmd/login.go]: That was my bad :), my cli test cmd was inverted :) this is what you get trying to send a patch during the holiday periods ;)

**Comment by chmouel** [comment]
@xcoulon yep sure, let's make it implicit if not specified otherwise!

File: Gopkg.toml (hunk 1 above)
Comment (xcoulon) [reply Gopkg.toml]: ah, ok, so maybe it's time to promote it to the `fabric8-services` org, I believe ;)

Comment (chmouel) [reply Gopkg.toml]: ah sounds perfect :)

**Review by xcoulon — APPROVED** [review]

**Comment by xcoulon** [comment]
thanks for this contrib, @chmouel ! 💪🙌

**Pull request merged**